### ISSUE TITLE
DML discrete outcome

### ISSUE THREAD

**Issue by nicolavizioli** [py-why/EconML#881]
Hi,
I saw that now DML (I'm using Linear) gives the possibility to select "discrete outcome=True", this means than now residuals of outcome model are not more only {-1,0,1} and then I can use a classifier like XGBClassifier() for outcome model? 
thanks

**Comment by kbattocchi** [comment]
That's correct - if you set `discrete_outcome=True` then your y_model should be an sklearn classifier rather than a regressor and we'll use `predict_proba` rather than `predict`, so the residuals can take on any values in the range [-1,1].

**Comment by brandongreenwell-8451** [comment]
Hi @kbattocchi and others! The use of the response residuals `y - p` seems problematic to me for a couple of reasons (their discreteness being one of them). Is there any theoretical justification for there use in the LinearDML approach? Can't seem to find anything in the literature and documentation.

**Comment by kbattocchi** [comment]
@brandongreenwell-8451 I don't think discreteness of the outcome should pose any particular problem - I believe the theory just requires that you can model E[Y|X,W], which is what using `predict_proba` with a classifier should give you (see https://econml.azurewebsites.net/spec/estimation/dml.html#overview-of-formal-methodology).

**Comment by brandongreenwell-8451** [comment]
Thanks @kbattocchi, I'll take another pass through the docs. Any idea how to interpret the result in this case?
